### DISCUSSION TITLE
TensorFlow is not linked to protobuf

### DISCUSSION THREAD

**Discussion by yl960122** [deepmodeling/deepmd-kit#3808] [Q&A]
### Summary

I encountered the error 'TensorFlow is not linked to protobuf' while compiling DeepMD, but in fact, I did not have the protobuf option when building TensorFlow==2.7.0 automatically. It seems that Protobuf is already built into libtensorflow_cc.so. I want to know how to resolve this error.
![image](https://github.com/deepmodeling/deepmd-kit/assets/65180595/59e54e15-54ed-4d3d-9c10-b507932ad927)
![image](https://github.com/deepmodeling/deepmd-kit/assets/65180595/8fdee02e-7029-4939-a9ca-49ebf0d9fc4a)

### DeePMD-kit Version

2.7.0

### TensorFlow Version

2.2.7

### Python Version, CUDA Version, GCC Version, LAMMPS Version, etc

cpu

### Details

I am compiling the CPU version of TensorFlow and DeepMDKit.

**Comment by yl960122** [comment]
The compilation command is:

cmake ../ -DTENSORFLOW_ROOT=~/software/tf \
-DCMAKE_INSTALL_PREFIX=~/software/deepmd-kit/source/dp_install \
-DLAMMPS_SOURCE_ROOT=~/software/lammps-2Aug2023 \
-DUSE_TF_PYTHON_LIBS=FALSE

**Comment by njzjz** [comment]
Protobuf headers are required, as shown in the message.
